src/internal/connector/sharepoint/list.go

```diff
@@ -0,0 +1,452 @@
+package sharepoint
+
+import (
+	"context"
+
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/microsoftgraph/msgraph-sdk-go/sites/item/contenttypes/item/basetypes"
+	"github.com/microsoftgraph/msgraph-sdk-go/sites/item/contenttypes/item/columnlinks"
+	"github.com/microsoftgraph/msgraph-sdk-go/sites/item/contenttypes/item/columnpositions"
+	tc "github.com/microsoftgraph/msgraph-sdk-go/sites/item/contenttypes/item/columns"
+	"github.com/microsoftgraph/msgraph-sdk-go/sites/item/lists"
+	"github.com/microsoftgraph/msgraph-sdk-go/sites/item/lists/item/columns"
+	"github.com/microsoftgraph/msgraph-sdk-go/sites/item/lists/item/contenttypes"
+	"github.com/microsoftgraph/msgraph-sdk-go/sites/item/lists/item/items"
+	"github.com/pkg/errors"
+
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/internal/connector/support"
+)
+
+// list.go contains additional functions to help retrieve SharePoint List data from M365
+// SharePoint lists represent lists on a site. Inherits additional properties from
+// baseItem: https://learn.microsoft.com/en-us/graph/api/resources/baseitem?view=graph-rest-1.0
+// The full details concerning SharePoint Lists can
+// be found at: https://learn.microsoft.com/en-us/graph/api/resources/list?view=graph-rest-1.0
+// Note additional calls are required for the relationships that exist outside of the object properties.
+
+// loadLists is a utility function to populate the List object.
+// @param identifier the M365 ID that represents the SharePoint Site
+// Makes additional calls to retrieve the following relationships:
+// - Columns
+// - ContentTypes
+// - List Items
+func loadLists(
+	ctx context.Context,
+	gs graph.Service,
+	identifier string,
```

Comment:
Nit, throughout this file:
```suggestion
	id string,
```

Reply:
Resolved as of [6e7fdb2](https://github.com/alcionai/corso/pull/1633/commits/6e7fdb245276f7ff194e8d6e1e37f79110246330)